File: app/src/main/java/eu/napcode/gonoteit/dao/NoteEntity.java
package eu.napcode.gonoteit.dao;

import android.arch.persistence.room.ColumnInfo;
import android.arch.persistence.room.Entity;
import android.arch.persistence.room.PrimaryKey;
import android.support.annotation.NonNull;

import javax.annotation.Nullable;

import eu.napcode.gonoteit.model.note.NoteModel;

import static eu.napcode.gonoteit.dao.NoteEntity.TABLE_NAME;

@Entity(tableName = TABLE_NAME)
public class NoteEntity {

    public static final String TABLE_NAME = "notes";

    public static final String COLUMN_UUID = "UUID";
    public static final String COLUMN_ID = "ID";
    public static final String COLUMN_TITLE = "title";
    public static final String COLUMN_CONTENT = "content";
    public static final String COLUMN_IMAGE = "image";
    public static final String COLUMN_UPDATED_AT = "updated";

    public NoteEntity() {
    }

    public NoteEntity(NoteModel noteModel) {
        this.setUuid(noteModel.getUuid());
        this.setTitle(noteModel.getTitle());
        this.setContent(noteModel.getContent());
        this.setId(noteModel.getId());
        this.setImageBase64(noteModel.getImageBase64());
        this.setUpdatedAt(noteModel.getUpdatedAt());
    }

    @NonNull
    @PrimaryKey
    @ColumnInfo(name = COLUMN_UUID)
    String uuid;

    @NonNull
    @ColumnInfo(name = COLUMN_ID)
    Long id;

    @Nullable
    @ColumnInfo(name = COLUMN_TITLE)
    private String title;

    @Nullable
    @ColumnInfo(name = COLUMN_CONTENT)
    private String content;

    @Nullable
    @ColumnInfo(name = COLUMN_IMAGE)
    private String imageBase64;

    @ColumnInfo(name = COLUMN_UPDATED_AT)
    private Long updatedAt;

    public String getUuid() {
        return uuid;
    }

    public void setUuid(String uuid) {
        this.uuid = uuid;
    }

    @NonNull
    public Long getId() {
        return id;
    }

    public void setId(@NonNull Long id) {
        this.id = id;
    }

    @Nullable
    public String getTitle() {
        return title;
    }

    public void setTitle(@Nullable String title) {
        this.title = title;
    }

    @Nullable
    public String getContent() {
        return content;
    }

    public void setContent(@Nullable String content) {
        this.content = content;
    }

    public void setImageBase64(@Nullable String imageBase64) {
        this.imageBase64 = imageBase64;
    }

    public String getImageBase64() {
        return imageBase64;
    }

    public Long getUpdatedAt() {
        return updatedAt;
    }

    public void setUpdatedAt(Long updatedAt) {
        this.updatedAt = updatedAt;
    }
}

File: app/src/main/java/eu/napcode/gonoteit/dao/NoteEntity.kt
package eu.napcode.gonoteit.dao

import android.arch.persistence.room.ColumnInfo
import android.arch.persistence.room.Entity
import android.arch.persistence.room.PrimaryKey
import android.arch.persistence.room.TypeConverters
import android.support.annotation.NonNull
import eu.napcode.gonoteit.dao.NoteEntity.Companion.TABLE_NAME

import eu.napcode.gonoteit.model.note.NoteModel
import eu.napcode.gonoteit.type.ReadAccess
import eu.napcode.gonoteit.type.WriteAccess
import eu.napcode.gonoteit.utils.ReadAccessConverter
import eu.napcode.gonoteit.utils.WriteAccessConverter

@Entity(tableName = TABLE_NAME)
data class NoteEntity(

        @PrimaryKey
        @NonNull
        @ColumnInfo(name = COLUMN_UUID)
        var uuid: String,

        @ColumnInfo(name = COLUMN_ID)
        var id: Long,

        @ColumnInfo(name = COLUMN_TITLE)
        var title: String?,

        @ColumnInfo(name = COLUMN_CONTENT)
        var content: String?,

        @ColumnInfo(name = COLUMN_IMAGE)
        var imageBase64: String?,

        @ColumnInfo(name = COLUMN_UPDATED_AT)
        var updatedAt: Long,

        @TypeConverters(ReadAccessConverter::class)
        @ColumnInfo(name = COLUMN_READ_PERMS)
        var readAccess : ReadAccess,

        @TypeConverters(WriteAccessConverter::class)
        @ColumnInfo(name = COLUMN_WRITE_PERMS)
        var writeAccess: WriteAccess
) {

    constructor(noteModel: NoteModel) :
            this(
                    uuid = noteModel.uuid!!,
                    id = noteModel.id!!,
                    updatedAt = noteModel.updatedAt!!,
                    title = noteModel.title,
                    content = noteModel.content,
                    imageBase64 = noteModel.imageBase64,
                    readAccess= noteModel.readAccess,
                    writeAccess= noteModel.writeAccess)


    companion object {

        const val TABLE_NAME = "notes"

        const val COLUMN_UUID = "UUID"
        const val COLUMN_ID = "ID"
        const val COLUMN_TITLE = "title"
        const val COLUMN_CONTENT = "content"
        const val COLUMN_IMAGE = "image"
        const val COLUMN_UPDATED_AT = "updated"
        const val COLUMN_READ_PERMS = "read"
        const val COLUMN_WRITE_PERMS = "write"
    }
}
